Library/Homebrew/extend/pathname.rb

```diff
@@ -392,7 +392,7 @@ def write_env_script(target, args, env = nil)
       args = nil
     end
     env_export = +""
-    env.each { |key, value| env_export << "#{key}=\"#{value}\" " }
+    env.each { |key, value| env_export << "#{key}=${#{key}-'#{value}'} " }
```

Comment:
Which formulae do you think this change would be useful for?

I think this breaks calls to `write_env_script` where I have
```ruby
env = { PATH: "/path/to/some/bin:$PATH" }
```
More generally, I think allowing for overrides is already possible (and indeed already done) with appropriate calls to `write_env_script`.

Whether an override should be allowed or not is probably better determined at the formula, rather than by default.

Reply:
> Which formulae do you think this change would be useful for?

The motivation for this PR is my recent changes to [v2ray](https://github.com/Homebrew/homebrew-core/blob/master/Formula/v2ray.rb) which breaks a GUI making use of environment variables to define custom assets path. Well the original executable can still be found in `opt_exec/v2ray`, but it's not obvious for most of its users.

> I think this breaks calls to `write_env_script` where I have, for example,
> 
> ```ruby
> env = { PATH: "/path/to/some/bin:$PATH" }
> ```

Yeah, I didn't consider that. Maybe a new helper should be created to provide options like `makeWrapper`.

> allowing for overrides is already possible with appropriate calls to `write_env_script`

Sorry if I'm wrong, but the variables are quoted, so one can't use parameter substitution with `write_env_script`. Is there any other way to define override?

Reply:
Why can't I define an `env` like
```ruby
env = { FOO: "${FOO:-#{Formula["foo"].opt_share}}" }
```
which makes it so that `FOO` is used from the environment first, and if not, it becomes `Formula["foo"].opt_prefix`? Parameter expansion is allowed in double-quotes, so this should become something like
```
FOO=${FOO:-/usr/local/opt/foo/share}
```